app/pages/notifications/notification-section.jsx

```diff
@@ -117,6 +118,19 @@ export default class NotificationSection extends Component {
     });
   }
 
+  markAllRead() {
+    talkClient.put('/notifications/read', { section: this.props.section });
+    this.state.notificationData.forEach((data) => {
+      data.notification.update({ delivered: true });
+    });
+    this.setState({ unread: 0 });
```

Comment:
Looks like this is a bit nondeterministic in the unread check.  Sometimes the read request finishes before the unread check, sometimes vice versa.